src/vm/ceemain.cpp

```diff
@@ -2074,15 +2074,8 @@ DWORD WINAPI EEShutDownProcForSTAThread(LPVOID lpParameter)
     {
         action = eRudeExitProcess;
     }
-    UINT exitCode;
-    if (g_fWeOwnProcess)
```

Comment:
g_fWeOwnProcess is dead right? it's always false. remove it throughout.

Reply:
Also, we should do the EEShutDownProcForSTAThread dance only that caused the bug only if `CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_FinalizeOnShutdown)` is set (it is legacy config switch that we may remove at some point).  EEShutDownProcForSTAThread was a workaround for deadlocks caused by finalization on shutdown. We got rid of finalization on shutdown, and so we should not have these deadlocks anymore.